compiler/ballerina-parser/src/main/java/io/ballerinalang/compiler/internal/parser/BallerinaParser.java

```diff
@@ -9050,7 +9049,12 @@ private STNode parseTupleMemberTypeDescList() {
             if (tupleMemberRhs == null) {
                 break;
             }
-
+            if (tupleMemberRhs.kind == SyntaxKind.ELLIPSIS_TOKEN) {
+                typeDesc = typeDescList.remove(typeDescList.size() - 1);
+                typeDesc = STNodeFactory.createRestDescriptorNode(typeDesc, tupleMemberRhs);
+                typeDescList.add(typeDesc);
+                break;
```

Comment:
Without removing and adding, can we do this check and create the RestDescriptorNode, before add it to the list.

May be introduce a new method like parseTupleMember() and do it inside that method. Then here instead of calling `parseTypeDescriptorInternal`we can call that method. WDYT?

Reply:
Changed.